addons/sms/controllers/main.py

```diff
@@ -0,0 +1,51 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+import re
+from operator import itemgetter
+
+from odoo import http
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class SmsController(http.Controller):
+
+    @http.route('/sms/status', type='json', auth='public')
+    def update_sms_status(self, message_statuses):
+        """Receive a batch of delivery reports from IAP
+
+        :param message_statuses:
+            [
+                {
+                    'sms_status': status0,
+                    'uuids': [uuid00, uuid01, ...],
+                }, {
+                    'sms_status': status1,
+                    'uuids': [uuid10, uuid11, ...],
+                },
+                ...
+            ]
+        """
+        SmsSms_sudo = http.request.env['sms.sms'].sudo()
+        SmsTracker_sudo = http.request.env['sms.tracker'].sudo()
+        all_uuids = []
+        for uuids, iap_status in map(itemgetter('uuids', 'sms_status'), message_statuses):
+            # Basic checks to avoid unnecessary queries and allow debugging.
+            if (
+                not uuids
+                or not iap_status
+                or not re.match(r'^\w+$', iap_status)
+                or any(not re.match(r'^[0-9a-f]{32}$', uuid) for uuid in uuids)
+            ):
```

Comment:
:socks:  Well, don't do too much formatting, we have some space on our screens quand même ^^
```suggestion
            if (
                not uuids or not iap_status or not re.match(r'^\w+$', iap_status)
                or any(not re.match(r'^[0-9a-f]{32}$', uuid) for uuid in uuids)
            ):
```

Reply:
It feels less readable (how many conditions are there?) but that's maybe personal, so it's done!